geode-core/src/main/java/org/apache/geode/cache/query/internal/QCompiler.java

```diff
@@ -200,6 +199,22 @@ public void compileImports(String imports) {
     Assert.assertTrue(stackSize() == 0, "stack size = " + stackSize() + ";stack=" + this.stack);
   }
 
+  private void checkWhereClauseForAggregates(CompiledValue compiledValue) {
+    if (compiledValue instanceof CompiledAggregateFunction) {
+      throw new QueryInvalidException(
+          "Aggregate functions can not be used as part of the criteria.");
```

Comment:
Specify the criteria - "where clause"
Change it to : "Aggregate functions are not supported with where clause"

Reply:
👍